src/lib/flows/share/steps/share-url.svelte

```diff
@@ -0,0 +1,213 @@
+<script lang="ts">
+  import StepLayout from '$lib/components/step-layout/step-layout.svelte';
+  import Button from '$lib/components/button/button.svelte';
+  import DownloadIcon from '$lib/components/icons/Download.svelte';
+  import downloadUrl from '$lib/utils/download-url';
+  import LinkIcon from '$lib/components/icons/Link.svelte';
+  import XIcon from '$lib/components/icons/X.svelte';
+  import FacebookIcon from '$lib/components/icons/Facebook.svelte';
+  import ThreadsIcon from '$lib/components/icons/Threads.svelte';
+  import TelegramIcon from '$lib/components/icons/Telegram.svelte';
+  import WhatsAppIcon from '$lib/components/icons/WhatsApp.svelte';
+  import EllipsisIcon from '$lib/components/icons/Ellipsis.svelte';
+  import CheckCircleIcon from '$lib/components/icons/CheckCircle.svelte';
+  import CopyIcon from '$lib/components/icons/Copy.svelte';
+  import EmailIcon from '$lib/components/icons/Email.svelte';
+  import { fade } from 'svelte/transition';
+  import { browser } from '$app/environment';
+  import { type ShareOption } from '../share-steps';
+
+  export let url: string = '';
+  export let downloadableImageUrl: string = '';
+  export let text: string = '';
```

Comment:
Small nitpick, but I think it'd be nicer to have these optional string props defined like this instead:

`export let prop: string | undefined = undefined` 

It communicates clearer that the prop is actually optional, and also makes it so that you don't have to explicitly pass an empty string when you actually want to pass nothing.

Reply:
Done!

Reply:
If URL is required, just remove the default — then it'll actually be treated as a required prop by the compiler

Reply:
Ah hah! Will update